### PR TITLE
8282947: JFR: Dump on shutdown live-locks in some conditions

### PR DESCRIPTION
**Pull request by judovana**
Reviewed-by: mgronlun

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8282947](https://bugs.openjdk.org/browse/JDK-8282947): JFR: Dump on shutdown live-locks in some conditions

### Reviewers
 * [Paul Hohensee](https://openjdk.org/census#phh) (@phohensee - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk8u-dev pull/234/head:pull/234` \
`$ git checkout pull/234`

Update a local copy of the PR: \
`$ git checkout pull/234` \
`$ git pull https://git.openjdk.org/jdk8u-dev pull/234/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 234`

View PR using the GUI difftool: \
`$ git pr show -t 234`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk8u-dev/pull/234.diff">https://git.openjdk.org/jdk8u-dev/pull/234.diff</a>

</details>

### CODE REVIEW

**Comment by phohensee** [comment]
Please change the PR title to "Backport 63eb0b7e8606dd9cd145e92eeeb744ff5b7be569".

**Comment by judovana** [comment]
Actually, the lib should be 
```
@library /lib /
```
?

**Comment by judovana** [comment]
Hmm. The test fails. I need to elaborate.

**Comment by judovana** [comment]
Hello! I had adapted the test. It sitll represents issue - fails on non patched jdk, and now correctly passes on patched jdk. Automated test is now running for both jfr group and jdk group. So far so good.

**Comment by judovana** [comment]
The pass can be visible on https://github.com/judovana/jdk-tester/actions/runs/3988734842/jobs/6840282234#step:12:187

What now?

**Comment by jerboaa** [comment]
> What now?

See `Contributing` section in https://wiki.openjdk.org/display/jdk8u

**Comment by judovana** [comment]
Thank you:) Label added. All testing on my end done. Waiting for review.

**Comment by zzambers** [comment]
I have tried to run jdk_jfr group on master, 3 failures are not regressions (fail both with and without changes):
https://github.com/zzambers/jdk-tester/actions/runs/3997510473/jobs/6858945097#step:12:2527

Failures are:
```
jdk/jfr/event/compiler/TestCompilerCompile.java
jdk/jfr/event/sampling/TestNative.java
jdk/jfr/jcmd/TestJcmdDumpPathToGCRoots.java
```

**Comment by judovana** [comment]
On my local machine only the `jdk/jfr/event/sampling/TestNative.java` is failing. I had fixed it like this: https://github.com/openjdk/jdk8u-dev/pull/235

**Comment by zzambers** [comment]
> On my local machine only the `jdk/jfr/event/sampling/TestNative.java` is failing. I had fixed it like this: #235

Interesting, so these 2 seem environment/test machine speed sensitive
TestCompilerCompile - timeout actually
TestJcmdDumpPathToGCRoots - ?

**Review by phohensee — APPROVED** [review]
Lgtm.

**Comment by jerboaa** [comment]
@judovana Please only integrate once you have approval via `jdk8u-fix-yes` on the bug.

**Comment by phohensee** [comment]
@judovana, you must add the jdk8u-fix-request label to the JBS issue, as well as a "Fix Request (8u)" comment detailing the testing and risk associated with the backport.

**Comment by judovana** [comment]
> @judovana, you must add the jdk8u-fix-request label to the JBS issue, as well as a "Fix Request (8u)" comment detailing the testing and risk associated with the backport.

The jdk8u-fix-request somehow disapeard. Readded. The  "Fix Request (8u)" is already there -  https://bugs.openjdk.org/browse/JDK-8282947?focusedCommentId=14554615&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-14554615

**Pull request closed without merging**
